src/nupic/bindings/algorithms.i

```diff
@@ -299,10 +299,18 @@ void forceRetentionOfImageSensorLiteLibrary(void) {
     self->load(inStream);
   }
 
-  inline void add_sample(float y_val, PyObject* x_vector)
+  inline PyObject* add_sample(float y_val, PyObject* x_vector)
   {
-    PyArrayObject* x = (PyArrayObject*) x_vector;
+    PyArrayObject* x = (PyArrayObject*)x_vector;
+    if (!PyArray_ISFLOAT(x) || PyArray_ITEMSIZE(x) != sizeof(float))
```

Comment:
I'm open to recommendations. AFAIK, `float` in C++ is always 32 bits but it would probably be better not to assume that. Since I don't really want to change functioality where possible, I could change the error message to not specify the bitness.

I am pretty sure that `PyArray_ISFLOAT` evaluates to true for both numpy.float32 and numpy.float64 but the C-style cast assumes that it is a `float` (32 bits usually). So I think I need the `ITEMSIZE` check as well. But let me know if you think otherwise before I repeat this check in a bunch of places!